From foo@baz Tue Apr  9 12:12:43 2002
Date: Wed, 13 Aug 2008 20:08:35 -0700
To: Greg KH <greg@kroah.com>
From: Greg Kroah-Hartman <gregkh@suse.de>
Subject: USB gadget: add f_ecm.c to g_util.ko

As f_ecm.c is also included in multiple .ko files, move it into
g_util.ko

Cc: David Brownell <david-b@pacbell.net>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/gadget/Makefile |    7 +++----
 drivers/usb/gadget/f_ecm.c  |    2 ++
 2 files changed, 5 insertions(+), 4 deletions(-)

--- a/drivers/usb/gadget/f_ecm.c
+++ b/drivers/usb/gadget/f_ecm.c
@@ -734,6 +734,7 @@ fail:
 
 	return status;
 }
+EXPORT_SYMBOL_GPL(ecm_bind);
 
 static void
 ecm_unbind(struct usb_configuration *c, struct usb_function *f)
@@ -829,3 +830,4 @@ int __init ecm_bind_config(struct usb_co
 	}
 	return status;
 }
+EXPORT_SYMBOL_GPL(ecm_bind_config);
--- a/drivers/usb/gadget/Makefile
+++ b/drivers/usb/gadget/Makefile
@@ -24,17 +24,16 @@ obj-$(CONFIG_USB_M66592)	+= m66592-udc.o
 #
 
 g_utils-objs			:= composite.o usbstring.o config.o \
-					epautoconf.o u_ether.o
+					epautoconf.o u_ether.o f_ecm.o
 
 g_zero-objs			:= zero.o f_sourcesink.o f_loopback.o
-g_ether-objs			:= ether.o f_subset.o f_ecm.o
+g_ether-objs			:= ether.o f_subset.o
 g_serial-objs			:= serial.o u_serial.o f_acm.o f_serial.o
 g_midi-objs			:= gmidi.o
 gadgetfs-objs			:= inode.o
 g_file_storage-objs		:= file_storage.o
 g_printer-objs			:= printer.o
-g_cdc-objs			:= cdc2.o f_ecm.o \
-					u_serial.o f_acm.o
+g_cdc-objs			:= cdc2.o u_serial.o f_acm.o
 
 ifeq ($(CONFIG_USB_ETH_RNDIS),y)
 	g_ether-objs		+= f_rndis.o rndis.o
